src/cosim/fmi/v2/fmu.cpp

```diff
@@ -63,6 +63,12 @@ fmu::fmu(
     modelDescription_.description = fmi2_import_get_description(handle_);
     modelDescription_.author = fmi2_import_get_author(handle_);
     modelDescription_.version = fmi2_import_get_model_version(handle_);
+    modelDescription_.capabilities.can_get_and_set_fmu_state = !!fmi2_import_get_capability(
+        handle_,
+        fmi2_cs_canGetAndSetFMUstate);
+    modelDescription_.capabilities.can_serialize_fmu_state = !!fmi2_import_get_capability(
+        handle_,
+        fmi2_cs_canSerializeFMUstate);
```

Comment:
Note that these capabilities are already checked in `slave_instance::copy_current_state()` and `slave_instance::export_state()`, respectively, and exceptions are thrown on any attempt to save/serialise state of FMUs that don't support it.

Is there a specific use case that makes it more advantageous to read them upfront and store them in the `model_description`? Or, put differently, when and how would one use `model_description::capabilities`?

Reply:
I thought it would be beneficial to have this information to the user before running the fmu instead of try and catch exception by actually running those functions. In this way we can inform users that the fmu(s) he/she is simulating support this capability and also help consumers of libcosim act accordingly.

Reply:
Ok, fair enough. Then I suggest rewriting the existing checks in `copy_current_state()` and `export_state()` so they use the new `model_description::capabilities` flags rather than re-calling the fmilib functions.

Reply:
Updated